ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -116,11 +119,20 @@ private[commands] class CommandTracker[Context](
           override def onPush(): Unit = {
             val submitRequest = grab(submitRequestIn)
             registerSubmission(submitRequest)
+            val commands = submitRequest.value.commands
+            val submissionId = commands.submissionId
+            val commandId = commands.commandId
             logger.trace(
-              "Submitted command {}",
-              submitRequest.value.commands.commandId,
+              "Submitted command {} in submission {}",
+              commandId,
+              submissionId,
+            )
+            push(
+              submitRequestOut,
+              submitRequest.enrich((context, _) =>
+                context -> TrackedCommandKey(submissionId, commandId)
```

Comment:
IMO we should group the submission IDs by change Id rather than command ID because deduplication happens on the `ChangeId` rather than the command ID alone.

Reply:
Sure. Previously it was the command ID alone, the story mentions adding just the submission ID. The change ID is: the submitting party (+ `act_as`) + application ID + command ID. What about the submission ID?

Reply:
The change ID is defined here: https://github.com/digital-asset/daml/blob/main/ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/ChangeId.scala

Sure. We should track the tuple (submissionId, changeId).

Reply:
There's no need to use change ID here, because there is a new `CommandTracker` for each (party, applicationId) tuple.

Reply:
Fair enough. I suppose that you mean for each `party union actAs`.

Reply:
Exactly, I forgot yesterday to double-check how the `Tracking.Key` looks like. It contains everything we need. Thanks!